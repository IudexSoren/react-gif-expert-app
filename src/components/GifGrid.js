import GifGridItem from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {

  const { data: images, loading } = useFetchGifs(category);

  return (
    <div className='card-grid'>
      <h3>{ category }</h3>
      { loading && <p className='animate__animated animate__flash animate__infinite'>Cargando...</p> }
      {
        images.map((img) => {
          return (
            <GifGridItem
              key={ img.id }
              { ...img }
            />
          )
        })
      }
    </div>
  )
}

export default GifGrid;