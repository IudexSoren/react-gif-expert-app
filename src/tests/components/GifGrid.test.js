import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba de GifGrid', () => {

  const category = 'One punch';

  test('Debe renderizar el componente correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={ category } />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar el componente con toda la informaciÃ³n cargada por useFetchGifs', () => {
    const gifs = [{
      id: 'ABC',
      title: 'TitlePrueba',
      url: 'UrlPrueba'
    },
    {
      id: 'DEF',
      title: 'TitlePrueba',
      url: 'UrlPrueba'
    }]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category={ category } />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });


});
