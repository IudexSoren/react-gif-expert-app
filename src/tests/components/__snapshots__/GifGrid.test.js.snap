// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifGrid Debe mostrar el componente con toda la informaci√≥n cargada por useFetchGifs 1`] = `
<div
  className="card-grid"
>
  <h3>
    One punch
  </h3>
  <GifGridItem
    id="ABC"
    key="ABC"
    title="TitlePrueba"
    url="UrlPrueba"
  />
  <GifGridItem
    id="DEF"
    key="DEF"
    title="TitlePrueba"
    url="UrlPrueba"
  />
</div>
`;

exports[`Prueba de GifGrid Debe renderizar el componente correctamente 1`] = `
<div
  className="card-grid"
>
  <h3>
    One punch
  </h3>
  <p
    className="animate__animated animate__flash animate__infinite"
  >
    Cargando...
  </p>
</div>
`;
